examples/lfric/eg16/create.py

```diff
@@ -0,0 +1,123 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2020, Science and Technology Facilities Council
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: R. W. Ford, STFC Daresbury Lab
+
+'''A simple Python script showing how to create an LFRic-specific
+PSyIR tree using the create methods. In order to use it you must first
+install PSyclone. See README.md in the top-level psyclone directory.
+
+Once you have psyclone installed, this script may be run by doing:
+
+>>> python create.py
+
+This should output a Fortran representation of the LFRic-PSyIR.
```

Comment:
If I run the example I get:

    (py3) eg16$ python create.py 
    module CONTAINER
    
      contains
      subroutine work(field2)
        use constants_mod, only : i_def, r_def
        real(kind=r_def), dimension(undf_w3), intent(inout) :: field2
        integer(kind=i_def) :: ndf_w3
        integer(kind=i_def) :: undf_w3
        ...

so `undf_w3` is declared locally, not passed as an argument and doesn't have a value. Also, its declaration comes *after* its use to dimension the array which the Intel compiler (used to) get upset about.

Reply:
Ah, I see. Sorry, I missed that you were talking about the example. I've made it so that all symbols are passed by argument apart from one field. This should now be OK. I've also ordered things so we get declarations of array sizes before the arrays are declared. However, this latter point is a general problem so I've created an issue for it #945.